posts/squidpy-essentials/squidpy-essentials.qmd

```diff
@@ -0,0 +1,1071 @@
+knitr::include_graphics("./figures/fig-export2.png")
+```
+
+# Install Squidpy {#sec-install}
+
+Assuming you have Python3 installed on your system, squidpy can be installed 
+using `pip3` like this: 
+
+```{default filename="Terminal"}
+#| eval: false
+pip3 install squidpy
+```
+
+For installing within a virtual environment (recommended), see the Appendix (@sec-venv).
+
+# Analysis without imaging data {#sec-noimage}
+
+In the simplest form, we can use squidpy without imaging data by reading
+the expression data, metadata, and optionally the FOV positions data. These analyses
+include normalization, PCA, umap, _etc._ I'll provide some code here but for more details please see the squidpy [vignette](https://squidpy.readthedocs.io/en/stable/notebooks/tutorials/tutorial_nanostring.html#analyze-nanostring-data){target="_blank"}.
+
+## Reading expression and metadata only
+
+Create an anndata object like this:
+```{filename="Python"}
+#| eval: false
+
+from pathlib import Path
+import os
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+import scanpy as sc
+import squidpy as sq
+
+flat_file_dir = 'path/to/breast_cancer_example/flatFiles/AUG29_13INTEGR_6K_BRST_PS_S2'
+meta_file = [item for item in os.listdir(flat_file_dir) if 'metadata_file' in item][0]
+counts_file = [item for item in os.listdir(flat_file_dir) if 'exprMat_file' in item][0]
+
+adata0 = sq.read.nanostring(
+    path=flat_file_dir,
+    counts_file=counts_file,
+    meta_file=meta_file
+)
+adata0
+```
+
+```
+AnnData object with n_obs × n_vars = 80073 × 6524
+    obs: 'RNA_nbclust_132d0e1b.dc7d.48de.814b.a88cd8d14f03_1_clusters', 'RNA_nbclust_132d0e1b.dc7d.48de.814b.a88cd8d14f03_1_posterior_probability', 'RNA_nbclust_9685fc9a.4f00.4267.99c1.ce1cb894807f_1_clusters', 'RNA_nbclust_9685fc9a.4f00.4267.99c1.ce1cb894807f_1_posterior_probability', 'cell', 'nCount_RNA', 'nFeature_RNA', 'nCount_negprobes', 'nFeature_negprobes', 'fov', 'Area', 'AspectRatio', 'Width', 'Height', 'Mean.PanCK', 'Max.PanCK', 'Mean.CD68_CK8_18', 'Max.CD68_CK8_18', 'Mean.CD298_B2M', 'Max.CD298_B2M', 'Mean.CD45', 'Max.CD45', 'Mean.DAPI', 'Max.DAPI', 'cell_id', 'assay_type', 'version', 'Run_Tissue_name', 'Panel', 'cellSegmentationSetId', 'cellSegmentationSetName', 'slide_ID', 'CenterX_global_px', 'CenterY_global_px', 'unassignedTranscripts', 'nCount_falsecode', 'nFeature_falsecode', 'Area.um2', 'propNegative', 'complexity', 'errorCtEstimate', 'percOfDataFromError', 'qcFlagsRNACounts', 'qcFlagsCellCounts', 'qcFlagsCellPropNeg', 'qcFlagsCellComplex', 'qcFlagsCellArea', 'qcCellsFlagged', 'median_negprobes', 'negprobes_quantile_0.9', 'median_RNA', 'RNA_quantile_0.9', 'nCell', 'nCount', 'nCountPerCell', 'nFeaturePerCell', 'propNegativeCellAvg', 'complexityCellAvg', 'errorCtPerCellEstimate', 'percOfDataFromErrorPerCell', 'qcFlagsFOV', 'i.median_negprobes', 'i.negprobes_quantile_0.9', 'i.median_RNA', 'i.RNA_quantile_0.9', 'cell_ID'
```

Comment:
One of the points is that the "flat files" that one downloads from AtoMx is not raw data and so it will have columns added to it from the analysis. I'm not sure the specifics of each of these particular columns.

Reply:
Just looking at this again. Explaining the individual columns of this particular anndata set is beyond the scope of this post.

Reply:
Take a look at the note and let me know what you think: a9420997d752c0a99f5112e6ad7a98417bc1587c